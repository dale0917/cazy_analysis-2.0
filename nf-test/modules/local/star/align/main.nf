process STAR_ALIGN {
    tag "$meta.id"
    label 'process_high'

    // conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:ded3841da0194af2701c780e9b3d653a85d27489-0' :
        'biocontainers/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:ded3841da0194af2701c780e9b3d653a85d27489-0' }"

    input:
    tuple val(meta), path(reads)
    path index

    output:
    tuple val(meta), path('*.Aligned.toTranscriptome.out.bam'), emit: bam
    tuple val(meta), path('*.Log.final.out')                  , emit: log_final
    path  "versions.yml"                                      , emit: versions
    path "*"                                                  , emit: other

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    STAR \\
        --runThreadN $task.cpus \\
        --genomeDir $index \\
        --outFileNamePrefix $prefix. \\
		--readFilesIn $reads \\
		--outSAMtype BAM SortedByCoordinate \\
        $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        star: \$(STAR --version | sed -e "s/STAR_//g")
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}Xd.out.bam
    touch ${prefix}.Log.final.out
    touch ${prefix}.Log.out
    touch ${prefix}.Log.progress.out
    touch ${prefix}.sortedByCoord.out.bam
    touch ${prefix}.toTranscriptome.out.bam
    touch ${prefix}.Aligned.unsort.out.bam
    touch ${prefix}.unmapped_1.fastq.gz
    touch ${prefix}.unmapped_2.fastq.gz
    touch ${prefix}.tab
    touch ${prefix}.Chimeric.out.junction
    touch ${prefix}.out.sam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        star: \$(STAR --version | sed -e "s/STAR_//g")
    END_VERSIONS
    """
}

