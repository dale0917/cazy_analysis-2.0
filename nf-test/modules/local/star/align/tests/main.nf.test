// zs modules test star/align
nextflow_process {

    name "Test Process STAR_ALIGN"
    script "../main.nf"
    config "./nextflow.config"
    process "STAR_ALIGN"

    tag "modules"
    tag "star"
    tag "star/align"

    setup {
        run("STAR_GENOMEGENERATE") {
            script "../../genomegenerate/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
                ])
                input[1] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true) ]
                ])
                """
            }
        }
    }

    test("star align paired end") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'nfcore-test-sample', single_end:false ], // meta map
                    [
                        file(params.test_data['homo_sapiens']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                    ]
                ]).view()
                input[1] = STAR_GENOMEGENERATE.out.index.map{ it[1] } // our star/align module takes only the index and not the tuple
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                        workflow,
                        path( "${outputDir}/nfcore-test-sample.Log.progress.out" ),
                        path( "${outputDir}/nfcore-test-sample.ReadsPerGene.out.tab" ),
                        path( "${outputDir}/nfcore-test-sample.SJ.out.tab" )
                    ).match() // these are the only ones deterministic
                }
            )
        }

    }

}
