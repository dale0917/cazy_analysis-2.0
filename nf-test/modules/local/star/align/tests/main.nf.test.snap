{
    "star align paired end": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "[[id:nfcore-test-sample, single_end:false], [/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz, /nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz]]"
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 2,
                    "tasksSucceeded": 2
                },
                "name": "workflow",
                "success": true
            },
            "nfcore-test-sample.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8",
            "nfcore-test-sample.ReadsPerGene.out.tab:md5,e14eef7c99a7f6fc7900959c4872ffb9",
            "nfcore-test-sample.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "timestamp": "2024-03-14T10:23:08.276894"
    }
}